kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/players/BaseAudioPlayer.kt

```diff
@@ -243,6 +243,11 @@ abstract class BaseAudioPlayer internal constructor(private val context: Context
     }
 
     fun play() {
+        if (exoPlayer.playbackState == Player.STATE_IDLE) {
+            exoPlayer.prepare();
+        } else if (exoPlayer.playbackState == Player.STATE_ENDED) {
+            exoPlayer.seekToDefaultPosition(exoPlayer.currentMediaItemIndex);
+        }
```

Comment:
There is an option to enable the player to repeat tracks if its at the end, so this is not something we need.